torch/_inductor/runtime/triton_heuristics.py

```diff
@@ -1189,18 +1193,19 @@ def make_launcher(self) -> LauncherType:
                 *call_args,
             ]
 
-        exec(
-            f"""
-            def launcher({', '.join(def_args)}, grid, stream):
-                if callable(grid):
-                    grid_0, grid_1, grid_2 = grid(grid_meta)
-                else:
-                    grid_0, grid_1, grid_2 = grid
-                runner({', '.join(runner_args)})
-                return bin
-            """.lstrip(),
-            scope,
-        )
+        if "extra_launcher_args" in self.inductor_meta:
+            def_args.extend(self.inductor_meta["extra_launcher_args"])
+
+        grid = GridExpr.from_meta(self.inductor_meta, cfg)
+        lines = [
+            f"def launcher({', '.join(def_args)}, stream):",
```

Comment:
nit: A comment that shows what these lines actually look like for a sample grid would be helpful, it's a bit hard to follow.

Maybe it'd be easier if grid just had a `gen_code()` function that generated the prefix followed by grid_0 = .. lines?

Reply:
I included an example in the PR description.  Full lines are harder to port between Python/C++/eval.